docs/_edot-sdks/java/troubleshooting.md

```diff
@@ -7,4 +7,57 @@ parent: EDOT Java
 
 # Troubleshooting the EDOT Java Agent
 
-TODO
\ No newline at end of file
+The sections below are in the order you should follow, unless you have already identified the section you need.
+
+This guide assumes you have tested the other components in the route from application+agent to Elastic Observability (eg collector or APM server, Elasticsearch, and Kibana) and that the problem has been isolated to the application+agent.
+
+## General
+
+Ensure you have set a service name (eg `-Dotel.service.name=Service1` or environment variable `OTEL_SERVICE_NAME` set to `Service1`) otherwise by default the data (traces, metrics, logs) will be sent to `unknown_service_java` - you may be getting data but it may all be under that service
+
+## Connectivity
+
+### To endpoint
+
+1. Check _from_ the host/VM/pod/container/image running the app, that connectivity is available to the APM server or collector. The examples here use a default URL, which you should replace with the endpoint you are using:
+
+- OpenTelemetry or EDOT collector: `curl -i http://127.0.0.1:4318/v1/traces -X POST -d '{}' -H content-type:application/json`
+- APM server on http: `curl --verbose -X GET http://127.0.0.1:8200`
+- APM server on https with secret token: `curl -X POST http://127.0.0.1:8200/ -H "Authorization: Bearer <secret_token>"`
+- APM server on https with API key: `curl -X POST http://127.0.0.1:8200/ -H "Authorization: ApiKey <api_key>"`
+
+The collector should produce output similar to
+```
+{"partialSuccess":{}}
+```
+
+The APM server should produce output similar to
+```
+{
+  "build_date": "2021-12-18T19:59:06Z",
+  "build_sha": "24fe620eeff5a19e2133c940c7e5ce1ceddb1445",
+  "publish_ready": true,
+  "version": "8.17.3"
+}
+```
+
+### Authentication
+
+In all cases, use the "APM server on https" endpoint connectivity test described above to confirm you are using the correct credentials
+
+## Is it the agent?
+
+Determine if the issue is related to the agent by
+
+1. Starting the application with no agent and seeing if the issue is not present, but then the issue is again present when restarting with the agent
```

Comment:
[minor] we can advise to disable the agent through configuration or env variable as well.

Reply:
yeah, but I've seen people think they have but haven't. Better to advise to remove it I think

Reply:
Good point, this is also something that can be easily verified on a running JVM by doing a `ps -ef|grep javaagent`, unlike setting an environment variable.